test/integration/image-optimizer/test/index.test.js

```diff
@@ -1045,4 +947,173 @@ describe('Image Optimizer', () => {
       await expectWidth(res, 8)
     })
   })
+
+  const sharpMissingText = `For production image optimization with Next.js, the optional 'sharp' package is strongly recommended`
+
+  const setupTests = (isSharp = false) => {
+    describe('dev support w/o next.config.js', () => {
+      let output = ''
+      const size = 384 // defaults defined in server/config.ts
+      beforeAll(async () => {
+        appPort = await findPort()
+        app = await launchApp(appDir, appPort, {
+          onStderr(msg) {
+            output += msg
+          },
+          cwd: appDir,
+        })
+      })
+      afterAll(async () => {
+        await killApp(app)
+        await fs.remove(imagesDir)
+      })
+
+      runTests({ w: size, isDev: true, domains: [] })
+
+      it('should not have sharp warning in development', () => {
+        expect(output).not.toContain(sharpMissingText)
+      })
+    })
+
+    describe('dev support with next.config.js', () => {
+      let output = ''
+      const size = 64
+      beforeAll(async () => {
+        const json = JSON.stringify({
+          images: {
+            deviceSizes: [largeSize],
+            imageSizes: [size],
+            domains,
+          },
+        })
+        nextConfig.replace('{ /* replaceme */ }', json)
+        appPort = await findPort()
+        app = await launchApp(appDir, appPort, {
+          onStderr(msg) {
+            output += msg
+          },
+          cwd: appDir,
+        })
+      })
+      afterAll(async () => {
+        await killApp(app)
+        nextConfig.restore()
+        await fs.remove(imagesDir)
+      })
+
+      runTests({ w: size, isDev: true, domains })
+
+      it('should not have sharp warning in development', () => {
+        expect(output).not.toContain(sharpMissingText)
+      })
```

Comment:
You could move this `it()` into the `runTests()` because we pass `isDev` 🙂 

Then you'll only need to define it once instead of for each of these `describe()`